### PR TITLE
Add support for 1984.is

### PR DESCRIPTION
**Pull request by so-lar-is**

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Has anyone else heard of this provider before?

**Comment by so-lar-is** [comment]
If you are worried about added maintenance burden, I am already using this code and will be maintaining it for the foreseeable future.

File: ddclient.in [hunk 1]
```diff
@@ -5198,6 +5208,81 @@ sub nic_freedns_update {
...
+sub nic_1984_examples {
+    return <<"EoEXAMPLE";
+
+o '1984'
+
+The '1984' protocol is used by DNS services offered by 1984.is.
+
+Configuration variables applicable to the '1984' protocol are:
+  protocol=1984                ##
+  password=service-password    ## your API key
```
Comment (SuperSandro2000) [inline ddclient.in:5223]:
```suggestion
  protocol=1984                ##
  password=api-key             ## your API key
```

File: ddclient.in [hunk 2]
```diff
@@ -5198,6 +5208,81 @@ sub nic_freedns_update {
...
+  fully.qualified.host         ## the domain to update
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=1984,                                                   \\
+  password=my-1984-api-key
```
Comment (SuperSandro2000) [inline ddclient.in:5229]:
```suggestion
  protocol=1984,                                                   \\
  password=my-1984-api-key,                                        \\
```

File: ddclient.in [hunk 3]
```diff
@@ -5198,6 +5208,81 @@ sub nic_freedns_update {
...
+        }
+        if ($response->{ok}) {
+            if ($response->{msg} =~ /unaltered/) {
+                success("Updating %s: skipped: IP was already set to %s", $host, $response->{ip});
+            } else {
+                success("%s -- Updated successfully to %s", $host, $response->{ip});
+            }
+        } else {
+            failed("%s", $response->{msg});
+        }
```
Comment (SuperSandro2000) [inline ddclient.in:5282]:
```suggestion
        unless ($response->{ok}) {
          failed("%s", $response->{msg});
        }

        if ($response->{msg} =~ /unaltered/) {
            success("Updating %s: skipped: IP was already set to %s", $host, $response->{ip});
        } else {
            success("%s -- Updated successfully to %s", $host, $response->{ip});
        }
```

**Review by SuperSandro2000 — COMMENTED** [review]
some small core improvements. Cutting the headers with \n\n may be not enough. Cloudflare recently received a bug fix about that.

**Comment by so-lar-is** [comment]
I have applied all enhancements including the improved header stripping.

**Pull request merged**